src/modules/EmailRecoveryModule.sol

```diff
@@ -0,0 +1,175 @@
+
+    event NewValidatorRecovery(address indexed validatorModule, bytes4 recoverySelector);
+
+    error NotTrustedRecoveryManager();
+    error InvalidSubjectParams();
+    error InvalidValidator(address validator);
+    error InvalidSelector(bytes4 selector);
+
+    mapping(address validatorModule => mapping(address account => bytes4 allowedSelector)) internal
+        allowedSelectors;
+    mapping(address account => address validator) internal validators;
+
+    constructor(address _zkEmailRecovery) {
+        emailRecoveryManager = _zkEmailRecovery;
+    }
+
+    modifier withoutUnsafeSelector(bytes4 recoverySelector) {
+        if (
+            recoverySelector == IModule.onUninstall.selector
+                || recoverySelector == IModule.onInstall.selector
+        ) {
+            revert InvalidSelector(recoverySelector);
+        }
+
+        _;
+    }
+
+    /*//////////////////////////////////////////////////////////////////////////
+                                     CONFIG
+    //////////////////////////////////////////////////////////////////////////*/
+
+    /**
+     * Initialize the module with the given data
+     * @param data The data to initialize the module with
+     */
+    function onInstall(bytes calldata data) external {
+        (
+            address validator,
+            bytes4 selector,
+            address[] memory guardians,
+            uint256[] memory weights,
+            uint256 threshold,
+            uint256 delay,
+            uint256 expiry
+        ) = abi.decode(data, (address, bytes4, address[], uint256[], uint256, uint256, uint256));
+
+        allowValidatorRecovery(validator, bytes("0"), selector);
+        validators[msg.sender] = validator;
+
+        _execute({
+            to: emailRecoveryManager,
+            value: 0,
+            data: abi.encodeCall(
+                IEmailRecoveryManager.configureRecovery,
+                (address(this), guardians, weights, threshold, delay, expiry)
+            )
+        });
+    }
+
+    function allowValidatorRecovery(
+        address validator,
+        bytes memory isInstalledContext,
+        bytes4 recoverySelector
+    )
+        internal
+        withoutUnsafeSelector(recoverySelector)
+    {
+        if (
+            !IERC7579Account(msg.sender).isModuleInstalled(
+                TYPE_VALIDATOR, validator, isInstalledContext
+            )
+        ) {
+            revert InvalidValidator(validator);
+        }
+
+        allowedSelectors[validator][msg.sender] = recoverySelector;
+
+        emit NewValidatorRecovery({ validatorModule: validator, recoverySelector: recoverySelector });
+    }
+
+    /**
+     * De-initialize the module with the given data
+     * @custom:unusedparam data - the data to de-initialize the module with
+     */
+    function onUninstall(bytes calldata /* data */ ) external {
+        address validator = validators[msg.sender];
+        delete allowedSelectors[validator][msg.sender];
```

Comment:
does this actually null out all the slots in the mapping? 
what would happen if the module is re-installed with a different allowed selector on the same validator module. both would be valid right?